### ISSUE TITLE
Window size & position are lost if I quit Zed in fullscreen

### ISSUE THREAD

**Issue by wzxu** [zed-industries/community#1700]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When I quit Zed in normal windowed mode and re-launch it, it remembers the last size & position just as expected. However when I quit Zed in fullscreen mode, it indeed reopens in fullscreen mode, but the previous normal window size & position are lost.

Steps to reproduce:
1. For easier observation, resize Zed to a smaller window and put it in the top right corner, or simply to a maximized size.
2. Quit & reopen Zed, and observe the window appears in previous size & position, as expected.
3. Go into fullscreen mode.
4. Quit & reopen Zed, and observe the window appears first in a small size in the lower left corner instead of previous position & size and then animates to fullscreen.
5. Go out of fullscreen mode, and observe the previous normal window state is lost and the window always defaults to a fixed size in the lower left corner. 

I seem to remember this worked correctly before so I think it's a regression in recent versions? Not really sure, but I only noticed it recently.

### Environment

Zed: v0.91.3 (stable)
OS: macOS 13.4.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by aesmail** [comment]
I have a similar issue where opening a new Zed window always opens it in fullscreen regardless of the size of the last window. It's a minor annoyance, but happens a lot.

### How to reproduce
cmd-shift-N (open a new window). It will open in fullscreen mode.

### Environment
Zed: 0.98.2
OS: macOS 13.4.1
Memory: 8GB and 16GB
Architecture: aarch64

<img width="400" alt="Screenshot 2023-08-15 at 09 59 52" src="https://github.com/zed-industries/community/assets/602354/f4509ea9-832e-4d1e-a26b-4f8cf22c73c7">

**Comment by kleinstein** [comment]
Zed window always opens it in fullscreen if i open a file with mouse double click in MacOS(Ventura) Finder.
 It annoying me very much.

**Comment by fuzenco** [comment]
> Zed window always opens it in fullscreen if i open a file with mouse double click in MacOS(Ventura) Finder. It annoying me very much.

Good catch. I didn’t realize that opening from File > Open does indeed retain the previous window size. I wish this was the case on double-clicking.

**Comment by michealp-coder** [comment]
I too have this problem.

I have no projects or files open. I simply launch Zed from Dock and it always goes full screen width/height.

Zed never persist the window size or location after close. Even for blank/new files.
